pkg/kubelet/kubelet.go

```diff
@@ -1159,6 +1192,7 @@ func (kl *Kubelet) SyncPods(allPods []*api.Pod, podSyncTypes map[types.UID]metri
 	//      though the pod was not considered terminated by the apiserver).
 	// These two conditions could be alleviated by checkpointing kubelet.
 	pods := kl.filterOutTerminatedPods(allPods)
+	pods = kl.filterOutPodsPastActiveDeadline(pods)
```

Comment:
A nit: could you follow the pattern similar to `handleNotFittingPods` and `handleOutofDisk` functions above (line 1179/1182)? They set the pods status to fail and let `filterOutTerminatedPods` handles all the filtering.